### PR TITLE
feat: setup of blogs_react app

### PR DESCRIPTION
**Pull request by Zain-Bajwa**
#### Blog app setup

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
None

#### What's this PR do?
Environment setup of blogs app

#### How should this be manually tested?
None

#### Any background context you want to provide?
None

#### Screenshots (if appropriate)
(Optional)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <Fragment>
+      {/* <h1>Hello</h1> */}
```
Comment (MAbdurrehman12) [inline src/App.js]: Please remove this comment

File: src/components/authentication/signIn.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import Typography from '@mui/material/Typography';
+import Container from '@mui/material/Container';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+/**
+ * 
+ * @param {*} [props] 
+ * @returns {ReactComponentElement} 
+ */
+const Copyright = (props) => {
```
Comment (MAbdurrehman12) [inline src/components/authentication/signIn.js]: I think this should be a separate component somewhere in common/shared folder. Copyright component inside signin component doesn't make much sense to me

File: src/components/authentication/signIn.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    console.log({
+      email: data.get('email'),
+      password: data.get('password'),
+    });
+  };
+
+  return (
+    <ThemeProvider theme={theme}>
```
Comment (MAbdurrehman12) [inline src/components/authentication/signIn.js]: I don't think this is the right place to use ThemeProvider.
We can wrap our complete app with themeProvider

File: src/components/authentication/signIn.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            alignItems: 'center',
+          }}
+        >
+          <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
+            <LockOutlinedIcon />
+          </Avatar>
+          <Typography component="h1" variant="h5">
+            Sign in
+          </Typography>
+          <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
```
Comment (MAbdurrehman12) [inline src/components/authentication/signIn.js]: Can we use fromik here?

File: src/utils/actionHandlers.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+        if (errorAction) {
+          dispatch(
+            errorAction({
+              error: error,
+              errorData,
+            })
+          );
+        }
+        // checking if return type is object then passing data to error callback
```
Comment (MAbdurrehman12) [inline src/utils/actionHandlers.js:65]: Can we remove these comments and appropriate code here?

File: src/utils/general.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+export const objectToQueryParams = (params) => {
```
Comment (MAbdurrehman12) [inline src/utils/general.js:9]: Please add function description (docstring) here

File: src/utils/general.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    Object.keys(params).forEach((key) => {
+      if (params[key] !== undefined && params[key] !== null) {
+        paramList.push(`${key}=${encodeURIComponent(params[key])}`);
+      }
+    });
+  
+    return paramList.join('&');
+  };
+  
+  export const getUrlWithParams = (requestUrl, params) => {
```
Comment (MAbdurrehman12) [inline src/utils/general.js]: Please add function description (docstring) here

**Review by MAbdurrehman12 — CHANGES_REQUESTED** [review]
Overall looks fine.
Just a couple of things mentioned

File: src/components/authentication/signIn.js (hunk 4 above)
Comment (Zain-Bajwa) [reply src/components/authentication/signIn.js]: At, this time I am not using `Formik` right now.

File: src/utils/actionHandlers.js (hunk 5 above)
Comment (Zain-Bajwa) [reply src/utils/actionHandlers.js:65]: We will remove these comments when we start writing our own logic. And these comments are generic.

**Review by MAbdurrehman12 — APPROVED** [review]
👍

**Pull request merged**